### PR TITLE
[Merged by Bors] - perf: improve some instances in GroupTheory/Congruence

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by ChrisHughes24** [comment]
I think this can wait until after the related Lean4 [issue](https://github.com/leanprover/lean4/issues/2451), and we can see if there is still a benefit to this.

File: Mathlib/GroupTheory/Congruence.lean [hunk 1]
```diff
@@ -1176,32 +1174,40 @@ instance {M : Type*} [Monoid M] (c : Con M) : Pow c.Quotient ℕ where
 @[to_additive "The quotient of an `AddSemigroup` by an additive congruence relation is
 an `AddSemigroup`."]
 instance semigroup {M : Type*} [Semigroup M] (c : Con M) : Semigroup c.Quotient :=
-  Function.Surjective.semigroup _ Quotient.surjective_Quotient_mk'' fun _ _ => rfl
```
Comment (mattrobball) [inline Mathlib/GroupTheory/Congruence.lean:1179]: I think a note explaining situation is good here. I would hope the slowdown in unification we see here (and in related files) can be fixed further up the file.  Similarly for the other explicit projections below.

**Review by mattrobball — APPROVED** [review]

**Comment by ChrisHughes24** [comment]
!bench

**Pull request closed without merging**